Modifies __init__.py to import properly

sabjorn opened the pull request:
On RPI3+ & Raspbian GNU/Linux 9 (stretch)
For whatever reason, when trying to use `pysimpledmx` I was unable to use `DMXConnection` recieving the error:

```
AttributeError: module 'pysimpledmx' has no attribute 'DMXConnection'
```

This commit changes the `__init__.py` to import "properly"(?). It is based on the python project Hitchiker Guide to Python [Structuring Your Project](http://docs.python-guide.org/en/latest/writing/structure/) page.

sabjorn committed:
Modifies __init__.py to import properly

On RPI3+ & Raspbian GNU/Linux 9 (stretch)
For whatever reason, when trying to use `pysimpledmx` I was unable to use `DMXConnection` recieving the error:

```
AttributeError: module 'pysimpledmx' has no attribute 'DMXConnection'
```

This commit changes the `__init__.py` to import "properly"(?). It is based on the python project Hitchiker Guide to Python (Structureing Your Project)[http://docs.python-guide.org/en/latest/writing/structure/] page.

sabjorn commented:
Might try to add some error handling to this in the future.

Would you consider doing a PR to `c0z3n/pySimpleDMX`? Or, possibly, requesting control of pip repo so that your Python3 mods can be used by more people?

JamesHannon commented:
Hi @sabjorn, I've not run into this issue myself I'll spin up a spare Pi and have a play after work

I didn't do a PR originally because the repo had gone quite, but I'm happy to message @c0z3n

sabjorn commented:
Hi @JamesHannon! Thanks for getting back to me!
Yeah, it's a weird situation for this import thing and i'm not sure why it's popping up. The change in this PR uses the formatting suggested [this](http://docs.python-guide.org/en/latest/writing/structure/) guide. So, while fixing my issues, hopefully it doesn't result in changes with the behaviour on your system.

c0z3n commented:
@JamesHannon Thanks for the mention! More than happy to take pull requests.

sabjorn committed:
modifies DMXConnection to allow error handling to propogate up out of module.

Specifically adds handling of serial connection during instatiation of DMXConnection. This error comes from the `serial` library and is emitted when the serial connection fails.

Adds a new member variable to DMXConnection, softfail, which allows the DMXConnection object to NOT exit the python application BUT also retains the original interface (`softfail` defaults to `False`).

sabjorn committed:
Fixes Channel 1 Problems

Channel 1 was not valid. This is now fixed.

sabjorn committed:
Adds exception handling

Exceptions created during instatiation of DMXConnection can now propigate out to main.

sabjorn committed:
Adds logger to class

sabjorn committed:
Modifies channel indexing to be [1, 511] inclusive

* modified setChannel to handle index of 1 -> 511.
* Added new error handler for this purpose, now calling value out of DMX channel range throws an error

sabjorn committed:
adds support for ramping multiple values DMX channels to specific values

sabjorn committed:
updated README to reflect new method

sabjorn commented:
@c0z3n I have recently added some new stuff to this project. Some error handling and a method for doing glide/ramping of DMX values (i.e. from current DMX value, transition to new value over S seconds).
Nothing i've added should break your original interface. The error handling is mostly to an alternative to the exit on failure when connecting to the device.

Are you interested in a PR of this work? If so, would you like me to break the PR into a few pieces?

@JamesHannon The work I've done to this project extends based on your mods to get it all to work for Python3. If I do a PR to @c0z3n it will, currently, also bring in your code. Are you okay with this?

sabjorn committed:
Modifies pysimpledmx to allow user set maximum number of DMX channels

The maximum number of DMX channels can be set explicitly. This allows of faster readout of data since all 512 channels don't have to be set on each `render()` call.

additionally:
* added VERSION file (setup.py now uses this file to set version number)
* update readme for clarity
* bumped version 0.2.0

sabjorn committed:
fix small typo in README

sabjorn committed:
Add hard system reboot for failing Serial communication

IMPORTANT: This is a hack
Adds a try/catch to com.write() in render() and a write timeout to serial connection. Now, if the com.write() times out (and a flag is set) the system is forced to reboot.

Note, this is all because the thread which runs the ramp() method cannot raise exceptions to the top level app.
The enttec device has a serial command which allows for a smooth transition between scenes, this would allow the removal of the thread and thus proper exception handling at the app level.

sabjorn commented:
@JamesHannon hi, any chance you can take a look at this?
